src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -93,11 +95,15 @@ protected void streamFeatures(HttpServletResponse response,
 	}
 
 	protected void streamBasin(HttpServletResponse response, String comid) throws Exception {
-		Map<String, Object> parameterMap = new HashMap<>();
-		parameterMap.put(Parameters.COMID, NumberUtils.parseNumber(comid, Integer.class));
-		BasinTransformer transformer = new BasinTransformer(response);
-		addContentHeader(response);
-		streamResults(transformer, BaseDao.BASIN, parameterMap);
+		if (comid != null) {
+			Map<String, Object> parameterMap = new HashMap<>();
+			parameterMap.put(Parameters.COMID, NumberUtils.parseNumber(comid, Integer.class));
+			BasinTransformer transformer = new BasinTransformer(response);
+			addContentHeader(response);
+			streamResults(transformer, BaseDao.BASIN, parameterMap);
+		} else {
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Basin was not found");
+		}
```

Comment:
Allowing a null comid through throws an IllegalArgumentException at line 100, which isn't specific enough to know what to do with later.  Thus, throwing a ResponseStatusException when it is null instead.

Reply:
I would leave this method alone. This is called from the LinkedDataController. The next method there (getFlowlines) has a better pattern for checking the comid.

Reply:
OK, I see that.  I didn't notice example in there before.